Sources/PointFree/Admin/Index.swift

```diff
@@ -62,29 +62,18 @@ private let newEpisodeEmailRowView = View<Episode> { ep in
     ])
 }
 
-let sendNewEpisodeEmailMiddleware: Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, Prelude.Unit>, Data> =
-  requireEpisode(notFoundMiddleware: redirect(to: .admin(.newEpisodeEmail(.show))))
-    <<< requireUser
+let sendNewEpisodeEmailMiddleware =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
     <<< requireAdmin
+    <<< filterMap(fetchEpisode >>> pure, or: redirect(to: .admin(.newEpisodeEmail(.show))))
     <| { conn in pure(conn.map(get2)) }
     >-> sendNewEpisodeEmails
     >-> redirect(to: .admin(.index))
 
-func requireEpisode<A>(
-  notFoundMiddleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, A>, Data>
-  )
-  -> (@escaping Middleware<StatusLineOpen, ResponseEnded, T2<Episode, A>, Data>)
-  -> Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, A>, Data> {
+func fetchEpisode(_ data: Tuple2<Database.User, Episode.Id>) -> Tuple2<Database.User, Episode>? {
```

Comment:
There's probably a lensier way of doing this? Maybe composing `over1` in the `filterMap` directly and have this just be `(Episode.Id) -> Episode?`?